### PR TITLE
[MU] Add support for POP3 in Twake mail

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Review by quantranhong1999 — APPROVED** [review]

**Comment by Arsnael** [comment]
memory server tests seem to crash for some reason?

**Review by Arsnael — APPROVED** [review]

**Comment by vttranlina** [comment]
> memory server tests seem to crash for some reason?

permission denied
maybe missing pop3 file

File: tmail-backend/apps/distributed/src/test/resources/pop3server.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0"?>
+<!-- Read https://james.apache.org/server/config-pop3.html for further details -->
+
+<pop3servers>
+    <pop3server enabled="false">
```
Comment (quantranhong1999) [inline tmail-backend/apps/distributed/src/test/resources/pop3server.xml:5]: enabled = false?

Comment (vttranlina) [reply tmail-backend/apps/distributed/src/test/resources/pop3server.xml:5]: yes, jut for CI pass
FOR [MU], just changing enabled=true

Comment (quantranhong1999) [reply tmail-backend/apps/distributed/src/test/resources/pop3server.xml:5]: > yes, jut for CI pass

Does it mean we need to add a dummy `pop3server.xml` file for every other deployments?

Comment (vttranlina) [reply tmail-backend/apps/distributed/src/test/resources/pop3server.xml:5]: hmm, maybe yep, but wait to CI result :P

Comment (quantranhong1999) [reply tmail-backend/apps/distributed/src/test/resources/pop3server.xml:5]: Better (I tried):

```
<pop3servers>
</pop3servers>
```

Likely there was a default POP3 configuration somewhere that tried to open directly for example the 995 port.

Comment (Arsnael) [reply tmail-backend/apps/distributed/src/test/resources/pop3server.xml:5]: I personnally like the module chooser idea, if we don't need pop3 for a deployment we just dont load it.

Comment (vttranlina) [reply tmail-backend/apps/distributed/src/test/resources/pop3server.xml:5]: Do we need contract tests for the POP3 Tmail server? 
It seems more complicated than the first commit. `Add support for POP3 in Twake mail`

File: tmail-backend/integration-tests/imap/memory-imap-integration-tests/src/test/resources/pop3server.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0"?>
+<pop3servers>
```
Comment (quantranhong1999) [inline tmail-backend/integration-tests/imap/memory-imap-integration-tests/src/test/resources/pop3server.xml:2]: With the module chooser now, I think the tests would pass even without the `pop3server.xml` file?

Comment (vttranlina) [reply tmail-backend/integration-tests/imap/memory-imap-integration-tests/src/test/resources/pop3server.xml:2]: No
I think it first too
but `pop3server.xml` still exist somewhere, (from james, when we inherit contract test from james)
Then still need override `pop3server.xml` file here

Comment (quantranhong1999) [reply tmail-backend/integration-tests/imap/memory-imap-integration-tests/src/test/resources/pop3server.xml:2]: It is this file: https://github.com/apache/james-project/blob/master/server/protocols/protocols-pop3/src/main/resources/pop3server.xml#L22-L21

Seems like a redundant file to me.

**Comment by Arsnael** [comment]
Jmap memory tests failing

**Comment by vttranlina** [comment]
green,

**Pull request merged**